main.py

```diff
@@ -74,15 +80,47 @@ def log(s):
   print('[' + str(datetime.now().strftime('%Y-%m-%dT%H:%M:%S')) + '] ' + s)
 
 
+def load_memory(memory_path, disable_bzip):
+  if disable_bzip:
+    with open(memory_path, 'rb') as pickle_file:
+      return pickle.load(pickle_file)
+
+  else:
+    with bz2.open(memory_path, 'rb') as zipped_pickle_file:
+      return pickle.load(zipped_pickle_file)
+
+
+def save_memory(memory, memory_path, disable_bzip):
+  if disable_bzip:
+    with open(memory_path, 'wb') as pickle_file:
+      pickle.dump(memory, pickle_file)
+
+  else:
+    with bz2.open(memory_path, 'wb') as zipped_pickle_file:
+      pickle.dump(memory, zipped_pickle_file)
+
+
 # Environment
 env = Env(args)
 env.train()
 action_space = env.action_space()
 
-
 # Agent
 dqn = Agent(args, env)
-mem = ReplayMemory(args, args.memory_capacity)
+
+# If a model is provided, and evaluate is fale, presumably we want to resume, so try to load memory
+if args.model is not None and not args.evaluate:
+  if not args.memory:
+    raise ValueError('Cannot resume training without memory save path. Aborting...')
+
```

Comment:
Can remove space and make this an `elif`.

Reply:
It's irrelevant since the condition in the first if raises an error, so the second if will never be executed anyway, but I did it.